### ISSUE TITLE
PE import error: Specified length extends beyond file bytes length

### ISSUE THREAD

**Issue by DCNick3** [NationalSecurityAgency/ghidra#2496]
**Describe the bug**
Ghidra is unable to import minimal file created with msvc with `/NOSTDLIB` flag.

**To Reproduce**
1. File > Import File
2. Select TestExe-msvc.exe
3. Try to import

**Expected behavior**
Ghidra is expected to import the file and proceed to disassemble it.

**Screenshots**
```
Specified length extends beyond file bytes length
java.lang.IndexOutOfBoundsException: Specified length extends beyond file bytes length
	at ghidra.program.database.mem.MemoryMapDB.checkFileBytesRange(MemoryMapDB.java:688)
	at ghidra.program.database.mem.MemoryMapDB.createInitializedBlock(MemoryMapDB.java:651)
	at ghidra.app.util.MemoryBlockUtils.createInitializedBlock(MemoryBlockUtils.java:227)
	at ghidra.app.util.opinion.PeLoader.processMemoryBlocks(PeLoader.java:662)
	at ghidra.app.util.opinion.PeLoader.load(PeLoader.java:121)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.doLoad(AbstractLibrarySupportLoader.java:347)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.loadProgram(AbstractLibrarySupportLoader.java:83)
	at ghidra.app.util.opinion.AbstractProgramLoader.load(AbstractProgramLoader.java:112)
	at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:400)
	at ghidra.plugin.importer.ImporterDialog.lambda$okCallback$7(ImporterDialog.java:349)
	at ghidra.util.task.TaskLauncher$1.run(TaskLauncher.java:90)
	at ghidra.util.task.Task.monitoredRun(Task.java:124)
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$1(TaskRunner.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

---------------------------------------------------
Build Date: 2020-Nov-15 1705 UTC
Ghidra Version: 9.2
Java Home: /usr/lib/jvm/java-11-openjdk
JVM Version: N/A 11.0.8
OS: Linux 5.8.14-arch1-1 amd64
Workstation: dcnick3-arch
```

**Attachments**
[TestExe-msvc.zip](https://github.com/NationalSecurityAgency/ghidra/files/5555677/TestExe-msvc.zip)

**Environment (please complete the following information):**
 - OS: Archlinux
 - Java Version: 11.0.8
 - Ghidra Version: 9.2

**Additional context**
Seems similar to #1869 and #1371, yet not working in 9.2.

**Comment by dev747368** [comment]
This file has an uninitialized .textbss section, which we were not handling correctly.  (we thought it was initialized and were trying to read the bytes from the file)

It appears that this section is usually present in binaries that are incrementally linked.  (including this detail for others searching for this issue)

**Comment by seritools** [comment]
I've got another executable with the same error message, the executable of the game MDK (official patch executable as acquired [here](https://soggi.org/misc/3dfx-voodoo-game-patches.htm)). It's a very old exe (1997, Watcom compiler/linker), so maybe it sets some section values weirdly?

[MDK3DFX.zip](https://github.com/NationalSecurityAgency/ghidra/files/5675460/MDK3DFX.zip)

Looking into it, it set the `IMAGE_SCN_CNT_UNINITIALIZED_DATA` section flag. However, instead of setting the virtual size and raw size according to the [PE format desc](https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#section-table-section-headers) over on MSDN (setting the `VirtualSize` to the size of the section and setting the `SizeOfRawData` to zero), it does the reverse: setting the `VirtualSize` to zero and specifying the length in `SizeOfRawData` instead.

**Comment by dev747368** [comment]
The original issue with the uninitialized section was fixed in bab1643b4fecf73f363568b8c6a7b53e8466910c.  @seritools , if this doesn't fix the file you mentioned, could you start a new issue and repeat the same info you put here?

**Comment by seritools** [comment]
@dev747368 just got around to checking it, my exe works now as well 👍
